cmd/copy.go

```diff
@@ -1282,16 +1283,25 @@ func (cca *CookedCopyCmdArgs) processRedirectionDownload(blobResource common.Res
 	}
 
 	// step 1: create client options
-	options := createClientOptions(common.LogNone)
+	options := &blockblob.ClientOptions{ClientOptions: createClientOptions(azcopyScanningLogger) }
 
 	// step 2: parse source url
 	u, err := blobResource.FullURL()
 	if err != nil {
 		return fmt.Errorf("fatal: cannot parse source blob URL due to error: %s", err.Error())
 	}
 
+	var blobClient *blockblob.Client
+	if credInfo.CredentialType.IsAzureOAuth() {
+		blobClient, err = blockblob.NewClient(u.String(), credInfo.OAuthTokenInfo.TokenCredential, options)
+	} else {
+		blobClient, err = blockblob.NewClientWithNoCredential(u.String(), options)
+	}
+	if err != nil {
+		return fmt.Errorf("fatal: Could not create client")
```

Comment:
Shall we not log the exact error that we received in err here as well, so that in debugging we get a clear clue on why client creation failed.